dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/filter/MetricsCollectExecutor.java

```diff
@@ -88,6 +90,10 @@ private void endExecute(Supplier<Boolean> rtStat){
     private void init(Invocation invocation) {
         String serviceUniqueName = invocation.getTargetServiceUniqueName();
         String methodName = invocation.getMethodName();
+        if (invocation instanceof RpcInvocation &&
+            isGenericCall(((RpcInvocation) invocation).getParameterTypesDesc(), methodName)) {
+            methodName = (String) invocation.getArguments()[0];
```

Comment:
When generalizing call, the number of parameters must be 3, but it is possible for the array to be out of bounds for some unknown reasons. I will add a check to avoid it.

Reply:
@CrazyHZM I solved it, please check the following.